tests/e2e/matmul/generate_e2e_matmul_tests.py

```diff
@@ -388,7 +388,7 @@ def get_test_compilation_infos(
     elif compilation_info_id == CompilationInfoId.SPIRVCooperativeMatrixVectorize:
         tile_workgroup_size_pairs = [
             TileWorkgroupSizePair(
-                [[64, 64], [16, 64], [0, 0, 16], [16, 16, 16]], [64, 4, 1]
+                [[64, 128], [32, 64], [0, 0, 32], [16, 16, 16]], [64, 2, 1]
             )
         ]
```

Comment:
As these e2e/matmul tests move out of iree-org/iree (https://github.com/iree-org/iree-test-suites/issues/2), we'll need to decouple this sort of test logic from the compiler [default] behavior.

As a general principle, tests (read: "users") should not be prying into implementation details, which these "test compilation infos" look like. We'll want a user-facing way to express any optimization hints or device properties.

Reply:
(I have a PR moving this file to the test suites repo here: https://github.com/iree-org/iree-test-suites/pull/13 and I'm going to find it hard to watch for diffs like this to backport)

Reply:
https://github.com/iree-org/iree/actions/runs/10461857158/job/28971377604?pr=18291#step:12:455
yeah, this setup is going to be tricky to support if we can't decouple the tests from the compiler

```
iree/iree/build/tests/matmul_vulkan_f16_large_rdna3_vulkan-spirv_vulkan_matmul.mlir:7:13: error: 'linalg.matmul' op expected workgroup x component equals to (warp_size * wg_tile_n / subgroup_tile_n)
  %result = linalg.matmul {compilation_info = #compilation0} ins(%lhs, %rhs: tensor<512x128xf16>, tensor<128x512xf16>) outs(%acc: tensor<512x512xf16>) -> tensor<512x512xf16>
            ^
```

(This is with my patches to move the tests out of tree)

Reply:
Agreed that it would be better for e2e tests to not expose these implementation details. Maybe we can track that as a separate issue since it would be a bigger effort?

I saw that you pulled the latest commit on to your conv/matmul tests PR, I hope that resolved the above issue.

Reply:
It resolves it for now, but once the tests live out of tree, any changes that need to cross the compiler and the tests will need simultaneous PRs to two repos. Treat this as a forcing function to define whatever compiler APIs are needed and rework the tests to use those APIs :)